src/containers/Subscription.js

```diff
@@ -4,32 +4,79 @@ import '../components/Auth/AuthModal.css';
 import { useKeycloak } from '@react-keycloak/web';
 import { useConfig } from '../hooks/useConfig';
 import axios from 'axios';
+import './Subscription.css';
 
 export const Subscription = () => {
   const { keycloak } = useKeycloak();
   const config = useConfig('/config/form.json');
   const userEmail = keycloak.idTokenParsed.email; 
   const userName = keycloak.idTokenParsed.given_name;
   const userToken = keycloak.token;
+  // segments is an array of objects {isChecked, segmentID, segmentName}
   const [ segments, setSegments ] = useState(null);
   const [ httpError, sethttpError] = useState(null);
+  const [ selectAll, setSelectAll] = useState(false);
 
-  const getformID = ( actionLink ) => {
-    return actionLink.charAt(actionLink.length-1)
+  const toggleCheckboxes = () => {
+    const toggledCheckedboxes = segments.map(({isChecked, ...others}) => ( { ...others, "isChecked": !selectAll } ));
+
+    setSegments(toggledCheckedboxes)
+    setSelectAll(!selectAll)
+  }
+
+  const createCheckboxes = () => {
+      return (
+        <div >
+          <div className="checkboxContent">
+            <input type ="checkbox" id="select_all" onChange={() => toggleCheckboxes()}/>
+            <label htmlFor="select_all">
+              Select all
+            </label>
+          </div>
+
+          <div className="checkboxContent">
+            {segments.map((contents, x) => (
+                <div key={contents.segmentID} className="checkboxContent"> 
+                  <input type ="checkbox" id ={contents.segmentID} checked={contents.isChecked} onChange={() => handleCheckbox(x)}/>
+                    <label htmlFor={contents.segmentID}>
+                      {contents.segmentName}, ischecked={String(contents.isChecked)}
+                    </label>
+                </div>
+            ))}
+          </div>
+        </div>
+      )
+  }
+
+  const handleCheckbox = (updateIndex) => {
+    const updatedSegments = [...segments]
+    updatedSegments[updateIndex].isChecked = !updatedSegments[updateIndex].isChecked
```

Comment:
a more clearn way of doing this is `segments.map(s => ({ ...s, isChecked: !s.isChecked})`

remember that `.map` returns a new array. When you do `[...segments]` you create a brand new array as well but it does not create new obejcts. those are all still pass by reference